pkg/remediator/watch/filteredwatcher.go

```diff
@@ -329,6 +341,8 @@ func (w *filteredWatcher) start(ctx context.Context, resourceVersion string) (bo
 	if err != nil {
 		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
 			return false, status.InternalWrapf(err, "failed to start watch for %s", w.gvk)
+		} else if apierrors.IsNotFound(err) {
+			metrics.RecordResourceConflict(ctx, w.latestCommit)
```

Comment:
should proabably use the getter here, and put a lock on the get/set methods to aoid race condition between the start and set methods.

Reply:
Does the start method block until the watch exits? If so, it can't use the same lock as the one protecting latestCommit.

This is partly why I don't love this approach. Multiple locks on a struct is a smell.

Reply:
Can you describe the race condition that you are concerned about?

Reply:
1. AddWatches for commit A starts a new filteredWatcher
2. AddWatches for commit B sets the latestCommit on the existing filteredWatcher at the same time the watcher errors and terminates the start method, causing a race between setting and getting the latestCommit field.

Reply:
I agree that stopping the watch for every commit seems wasteful, but `go test -race` will fail if the `latestCommit` field is read from and written to the same time from two different threads. So it needs locking around the field to allow concurrent use, even if it doesn't solve a specific failure scenario.

Reply:
Good point, done